core/src/main/java/org/apache/iceberg/FileCleanupStrategy.java

```diff
@@ -83,4 +85,33 @@ protected void deleteFiles(Set<String> pathsToDelete, String fileType) {
             (file, thrown) -> LOG.warn("Delete failed for {} file: {}", fileType, file, thrown))
         .run(deleteFunc::accept);
   }
+
+  protected Set<String> expiredStatisticsFilesLocations(
+      TableMetadata beforeExpiration, Set<Long> validIds, Set<Long> expiredIds) {
+    Set<String> validStatisticsFilesLocation = Sets.newHashSet();
+    for (long snapshotId : validIds) {
+      List<StatisticsFile> statisticsFiles =
+          beforeExpiration.statisticsFilesForSnapshot(snapshotId);
+      if (statisticsFiles != null) {
+        statisticsFiles.forEach(file -> validStatisticsFilesLocation.add(file.path()));
+      }
+    }
+
+    // Statistics files can be reused in case of rewrite data files.
```

Comment:
Why would rewrite data files be involved?

Reply:
I can remove/update the comment. 

But I just wanted to mention stats files can be reused (rewrite data files is one of the scenarios). So, we can't blindly remove it without checking the reachability graph.